cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -41,6 +41,7 @@ def _get_devices(self, label):
 
         label_list = util.find_devs_with("LABEL=%s" % label.upper())
         label_list.extend(util.find_devs_with("LABEL=%s" % label.lower()))
+        label_list.extend(util.find_devs_with("LABEL_FATBOOT=%s" % label))
```

Comment:
find_devs_with calls some *bsd implelementation functions... and I'm pretty sure that LABEL_FATBOOT wont work for them, i'm not sure if that matters.  but we shoudl be sure to not regress something there.

Reply:
This was added with an alternative. FAT_BOOT has been included in current systems as a new form. That's why I added it to the code. LABEL will continue to function.

Reply:
You need to at least make sure that passing 'LABEL_FATBOOT' through to the code for the 'find_devs_with_*bsd' functions (https://github.com/canonical/cloud-init/blob/995f8adf00509e5d2aefc9f0680c3c4894ae6666/cloudinit/util.py#L1112) doesn't cause a problem.  That code has never had LABEL_FATBOOT run through it before (It obviously *should not*, but please read and at least convince yourself that it wil not).

Reply:
I don't know how to fix this by editing this other file.

Reply:
@smoser, thank you for drawing attention to this

@goneri wrote most of the `find_devs_with_*bsd()` code, and he uses NoCloud, so perhaps he'll have some word on it

for now, the code is very naïve, and not written to be extensible to new classes of things to look for, with the justification that certain Linux specific things osten don't map under BSD, and what we had was good enough